lib/salad_ui/helpers.ex

```diff
@@ -83,4 +83,48 @@ defmodule SaladUI.Helpers do
         "#{align}-vertical"
     end
   end
+
+  # Translate error message
+  # borrowed from https://github.com/petalframework/petal_components/blob/main/lib/petal_components/field.ex#L414
+  defp translate_error({msg, opts}) do
+    config_translator = get_translator_from_config() || (&fallback_translate_error/1)
+
+    config_translator.({msg, opts})
+  end
+
+  defp fallback_translate_error({msg, opts}) do
+    Enum.reduce(opts, msg, fn {key, value}, acc ->
+      try do
+        String.replace(acc, "%{#{key}}", to_string(value))
+      rescue
+        e ->
+          IO.warn(
+            """
+            the fallback message translator for the form_field_error function cannot handle the given value.
+
+            Hint: you can set up the `error_translator_function` to route all errors to your application helpers:
+
+              config :salad_ui, :error_translator_function, {MyAppWeb.CoreComponents, :translate_error}
+
+            Given value: #{inspect(value)}
+
+            Exception: #{Exception.message(e)}
+            """,
+            __STACKTRACE__
+          )
+
+          "invalid value"
+      end
+    end)
+  end
+
+  defp get_translator_from_config do
+    case_result =
+      case Application.get_env(:salad_ui, :error_translator_function) do
+        {module, function} -> &apply(module, function, [&1])
+        nil -> nil
+      end
+
+    IO.inspect(case_result, label: "error_translator_function")
```

Comment:
Probably good to remove this `IO.inspect`. Otherwise it looks good.

Reply:
oh my bad, thank you